drivers/clock_control/nrf_clock_calibration.c

```diff
@@ -194,8 +198,14 @@ static void measure_temperature(struct k_work *work)
 	s16_t diff;
 	bool started = false;
 	int key;
+	int rc;
+
+	rc = get_temperature(&temperature);
+	if (rc != 0) {
```

Comment:
It seems to be that leaving here will break calibration algorithm as state machine will stay with HF clock turned on and nothing started to change the state (no calibration or calibration timer active). When calibration initially starts then `calib_skip_cnt` is set to 0 to ensure that calibration will be performed in first attempt disregarding temperature measurement. Initial temperature measurement is used only to have value for next iteration to calculate diff. If temperature sensor does not return value, then it would be better to set temperature to some arbitrary value (something unrealistic like 150'C). That will result in additional calibration during next timeout due to temperature change.

Reply:
I'm not going to try tricks with temperatures; instead I've updated it to reset `cal_state` to the state that it was in when the callback determined it was necessary to invoke `measure_temperature()`.  Does that work?  If not I'd rather reassign this to you so you get a solution you're willing to maintain.

Reply:
Sorry for the delay. I've reproduced the problem. The proposed solution stops calibration because state machine stuck in that state as there is no asynchronous event to push it forward. I still think that simplest and most reasonable is to set unrealistic temperature. That will force calibration.

Something like:
```
if (rc != 0) {
  temperature = INT16_MAX;
  LOG_WRN("Temperature sensor not ready. Setting max temperature to enforce calibration.");
}
```

Initially, i thought about requesting temperature measurement again (`k_work_submit(&temp_measure_work)`) but that won't work as this work submit has higher priority so calling it recursively leads to deadlock.